features/developer_portal/admin/account/personal_details.feature

```diff
@@ -0,0 +1,82 @@
+Feature: Dev Portal Buyer Personal Details
+  As a buyer
+  I want to change my personal details
+
+  Background:
+    Given a buyer logged in to a provider
+
+  # This is needed, otherwise assert_flash doesn't receive the right message.
+  # Check flash-buyer.js
+  @javascript
+  Scenario: Buyer doesn't use current password
+    Given the buyer wants to edit their personal details
+    When the buyer edits their personal details
+    Then they should not be able to edit their personal details
+
+  @javascript
+  Scenario: Buyer uses wrong current password
+    Given the buyer wants to edit their personal details
+    When the buyer edits their personal details
+    And the buyer writes a wrong current password
+    Then they should not be able to edit their personal details
+
+  @javascript
+  Scenario: Buyer uses correct current password
+    Given the buyer wants to edit their personal details
+    When the buyer edits their personal details
+    And they change their password
+    And the buyer writes a correct current password
+    Then they should be able to edit their personal details
+    And password must have changed
+
+  @javascript
+  Scenario: Provider has custom personal details fields
+    Given provider "foo.3scale.localhost" has the following fields defined for "User":
+      | name                 | required | read_only | hidden |
+      | first_name           | true     |           |        |
+      | last_name            |          | true      |        |
+      | job_role             |          |           | true   |
+      | user_extra_required  | true     |           |        |
+      | user_extra_read_only |          | true      |        |
+      | user_extra_hidden    |          |           | true   |
+
+    When the buyer wants to edit their personal details
+    Then fields should be required:
+      | required            |
+      | First name          |
+      | User extra required |
+
+    And they should not see the fields:
+      | not present          |
+      | Last name            |
+      | Job role             |
+      | User extra read only |
+      | User extra hidden    |
+
+    When the buyer writes a correct current password
+    And they press "Update Personal Details"
```

Comment:
This makes me think that current password is verified separately, whereas to me it would make for a better UX to verify everything at the same time. This way, when the password (or the wrong password) is provided, the form will render all errors at the same time: password wrong and missing fields.